Client/Card.ts

```diff
@@ -1,27 +1,37 @@
 import * as gracely from "gracely"
 import * as http from "cloudly-http"
-import * as rest from "cloudly-rest"
 import { Card as modelCard } from "../Card"
 
-export class Card extends rest.Collection<gracely.Error> {
+export class Card extends http.Client<gracely.Error> {
 	async detokenize(token: modelCard.Token): Promise<modelCard | string | gracely.Error> {
-		return await this.client.get<modelCard | string>(`card/${token}`)
+		return await this.get<modelCard | string>(`card/${token}`)
 	}
 	async getImage(token: modelCard.Token, cardHolderName?: string): Promise<string | gracely.Error> {
 		const queryString = cardHolderName ? `?ch=${cardHolderName}` : ""
-		return await this.client.get<string>(`card/${token}${queryString}`, { accept: ["image/svg+xml"] })
+		return await this.get<string>(`card/${token}${queryString}`, { accept: ["image/svg+xml"] })
 	}
 	async getPdf(token: modelCard.Token, cardHolderName?: string): Promise<string | gracely.Error> {
 		const queryString = cardHolderName ? `?ch=${cardHolderName}` : ""
-		return await this.client.get<string>(`card/${token}${queryString}`, { accept: ["application/pdf"] })
+		return await this.get<string>(`card/${token}${queryString}`, { accept: ["application/pdf"] })
+	}
+	async getImageWithRawData(card: modelCard, cardHolderName?: string): Promise<string | gracely.Error> {
+		const queryString = cardHolderName ? `?ch=${cardHolderName}` : ""
+		return await this.post<string>(`card/raw${queryString}`, card, { accept: ["image/svg+xml"] })
+	}
+	async getPdfWithRawData(card: modelCard, cardHolderName?: string): Promise<ArrayBuffer | gracely.Error> {
+		const queryString = cardHolderName ? `?ch=${cardHolderName}` : ""
```

Comment:
this line is copied and pasted 4 times and would be a lovely function

function prepareUrl(String basePath, String cardHolderName?) : String